src/index.ts

```diff
@@ -0,0 +1,323 @@
+
+export const handler = async (event: any = {}): Promise<any> => {
+  const startedAt = new Date();
+  console.log("[DEBUG]\tstarting beaconUpdate...");
+  // **************************************************************************
+  // 1. Load config (this file must be the same as the one used by the node)
+  // **************************************************************************
+  const nodeConfigPath = path.resolve(`${__dirname}/../config/airnode.json`);
+  const nodeConfig = node.config.parseConfig(nodeConfigPath, process.env);
+  const keeperConfig = loadAirkeeperConfig();
+  const config = merge(nodeConfig, keeperConfig);
+
+  const { chains, nodeSettings, triggers, ois, apiCredentials } = config;
+  const evmChains = chains.filter(
+    (chain: node.ChainConfig & ChainConfig) => chain.type === "evm"
+  );
+  if (isEmpty(chains)) {
+    throw new Error(
+      "One or more evm compatible chain(s) must be defined in the provided config"
+    );
+  }
+
+  const providerPromises = flatMap(
+    evmChains.map((chain: node.ChainConfig & ChainConfig) => {
+      return map(chain.providers, async (chainProvider, _) => {
+        // **************************************************************************
+        // 2. Initialize provider and contracts
+        // **************************************************************************
+        console.log("[DEBUG]\tinitializing...");
+        const chainProviderUrl = chainProvider.url || "";
+        const provider = buildEVMProvider(chainProviderUrl, chain.id);
+
+        const airnodeRrp = protocol.AirnodeRrpFactory.connect(
+          chain.contracts.AirnodeRrp,
+          provider
+        );
+
+        // TODO: use factory class to create contract instead
+        // const rrpBeaconServer = RrpBeaconServerFactory.connect(
+        //   chain.contracts.RrpBeaconServer,
+        //   provider
+        // );
+        const abi = RrpBeaconServer.abi;
+        const rrpBeaconServer = new ethers.Contract(
+          chain.contracts.RrpBeaconServer,
+          abi,
+          provider
+        );
+
+        // **************************************************************************
+        // 3. Run each keeper job
+        // **************************************************************************
+        for (const {
+          templateId,
+          oisTitle,
+          endpointName,
+          deviationPercentage,
+          keeperSponsor,
+          requestSponsor,
+        } of triggers.rrpBeaconServerKeeperJobs) {
+          // **************************************************************************
+          // 4. Fetch template by ID
+          // **************************************************************************
+          console.log("[DEBUG]\tfetching template...");
+          const template = await airnodeRrp.templates(templateId);
+          if (!template) {
+            console.log("[ERROR]\ttemplate not found");
+          }
+
```

Comment:
🤔 so how would that "look up" work? Airkeeper is currently iterating over the `triggers.rrpBeaconServerKeeperJobs` list so in order to get the `endpointName` and `oisTitle` it would have to also iterate over the `ois.endpoints` array for each template then derive the `endpointId` and compare it with the `endpointId` from the template? 

It can be done but it seems much efficient to just add `endpointName` and `oisTitle` to the triggers.rrpBeaconServerKeeperJobs object tbh.